ui/src/main/webapp/health-ui/healthui.js

```diff
@@ -0,0 +1,194 @@
+/* 
+ * Javascript file for health UI
+ */
+var refreshTimer;
+
+function processSettings() {
+
+    // Title
+    var title = localStorage.getItem("title");
+    if (title === null || title === '')
+        title = "Health UI";
+    $("#navbar_title").html(title);
+    $("#settings_form_title").val(title);
+    document.title = title;
+
+    // Poll
+    var poll = localStorage.getItem("poll");
+    $("#settings_form_poll").val(poll);
+    if (poll === null || poll === '' || poll === 'off') {
+        clearInterval(refreshTimer);
+    } else {
+        var interval = getInterval(poll);
+        clearInterval(refreshTimer);
+        if (interval > 0) {
+            refreshTimer = setInterval(function () {
+                loadHealthData();
+            }, interval);
+        }
+    }
+
+}
+
+function loadHealthData() {
+    var xmlhttp = new XMLHttpRequest();
+    var url = getUrl();
+    xmlhttp.onreadystatechange = function () {
+        if (xmlhttp.readyState === 4) {
+            if (xmlhttp.status === 200 || xmlhttp.status === 503) {
+                var healthprobes = JSON.parse(this.responseText);
+                processOk(healthprobes);
+            } else {
+                processError(xmlhttp);
+            }
+        }
+    };
+    xmlhttp.open("GET", url, true);
+    xmlhttp.send();
+}
+
+function processOk(healthprobes) {
+    var updown = healthprobes.outcome;
+        
+    if(updown == null){
+        updown = healthprobes.status; // Changed in later MP versions
```

Comment:
I believe this can be the only one now since we dropped "outcome" in the spec and SR is up to date. Same for state/status.